CodeGeneration/Sources/Utils/CodeGenerationFormat.swift

```diff
@@ -42,4 +42,71 @@ public class CodeGenerationFormat: BasicFormat {
       return super.visit(node)
     }
   }
+
+  public override func visit(_ node: TupleExprElementListSyntax) -> TupleExprElementListSyntax {
+    let children = node.children(viewMode: .all)
+    // If the function only takes a single argument, display it on the same line
+    if let callee = node.parent?.as(FunctionCallExprSyntax.self)?.calledExpression.as(MemberAccessExprSyntax.self), callee.base == nil {
+      // This is a constructor for tokens. Write them on a single line
+      return super.visit(node)
+    }
+    if children.count > 3 {
+      return TupleExprElementListSyntax(formatChildrenSeparatedByNewline(children: children, elementType: TupleExprElementSyntax.self))
+    } else {
+      return super.visit(node)
+    }
+  }
+
+  public override func visit(_ node: FunctionParameterListSyntax) -> FunctionParameterListSyntax {
+    let children = node.children(viewMode: .all)
+    // Short array literals are presented on one line, list each element on a different line.
+    if children.count > 3 {
+      return FunctionParameterListSyntax(formatChildrenSeparatedByNewline(children: children, elementType: FunctionParameterSyntax.self))
+    } else {
+      return super.visit(node)
+    }
+  }
+
+  public override func visit(_ node: ArrayElementListSyntax) -> ArrayElementListSyntax {
+    let children = node.children(viewMode: .all)
+    // Short array literals are presented on one line, list each element on a different line.
+    if children.count > 2 {
+      return ArrayElementListSyntax(formatChildrenSeparatedByNewline(children: children, elementType: ArrayElementSyntax.self))
+    } else {
+      return super.visit(node)
+    }
+  }
+
+  public override func visit(_ node: DictionaryElementListSyntax) -> DictionaryElementListSyntax {
+    let children = node.children(viewMode: .all)
+    // Short array literals are presented on one line, list each element on a different line.
+    if children.count > 2 {
+      return DictionaryElementListSyntax(formatChildrenSeparatedByNewline(children: children, elementType: DictionaryElementSyntax.self))
+    } else {
+      return super.visit(node)
+    }
+  }
+}
+
+// MARK: - Private helpers
+
+fileprivate extension CodeGenerationFormat {
+  func formatChildrenSeparatedByNewline<SyntaxType: SyntaxProtocol>(children: SyntaxChildren, elementType: SyntaxType.Type) -> [SyntaxType] {
```

Comment:
Is there any reason why you put this inside an extension and didn’t just add as a `private` function to the definition of `CodeGenerationFormat`?

Reply:
Oh, it's a habit of split the code. 

I'll move it.